api/hub-classes/index.js

```diff
@@ -73,8 +79,63 @@ exports.withAvailability = async function (n, HubClass, compiledQuery) {
   }, {
     $project
   }]);
+  const hubClassIds = hubClasses.map(({ _id }) => _id );  
+  const registrations = await bookedResources(undefined, HubRegistration, { query: { hubClass: { $in: hubClassIds } } });
+
+  function getGeneratedTimes (cl) {
+    const REPEATER = 8;
+
+    let nextDates = [];
+    let { dayRef, timeBlock } = cl.generateTimeRef;
+    if (dayRef) {
+      let today = moment().startOf('day');
+      let ref = recur(today).every(dayRef).daysOfWeek();
+
+      nextDates = ref.next(REPEATER).reduce((acc, cur, index) => {
+        let startString = `${moment(cur).format('MM DD YYYY')} ${timeBlock.start.hour}:${timeBlock.start.min}`;
+        let endString = `${moment(cur).format('MM DD YYYY')} ${timeBlock.end.hour}:${timeBlock.end.min}`;
+
+        let conflictTimes = (registrations || []).filter(registration => {
+          const classname = ((registration.hubClassInfo || [])[0] || {}).name || ((registration.classes || [])[0] || {}).name;
+          const isEssential = (!!classname.match(/safety essentials/ig) || {}).length;
+          const conflictTimeStart = moment(moment(cur).format('MM DD YYYY'), 'MM DD YYYY').hour(timeBlock.start.hour).minute(timeBlock.start.min);
+          
+          return conflictTimeStart.isSame(moment(registration.start)) && !isEssential;
+        });
+
+        const participantCountForTimeblock = conflictTimes.reduce((acc, { participants, trainee }) => {
+          let calc = trainee ? trainee.length : (participants || {}).length || 0;
+          let updatedCount = calc + acc;
+          return updatedCount;
+        }, 0);
+
+        if (participantCountForTimeblock >= cl.seats) {
+          return acc;
+        }
+
+        acc.push({
+          start: new Date(startString).toLocaleString('en', { timeZone: 'America/Denver' }),
+          end:   new Date(endString).toLocaleString('en', { timeZone: 'America/Denver' }),
+          _id:   index
+        });
+        return acc;
+      }, []);
+    }
+    return nextDates;
+  }
+
+  hubClasses = hubClasses.map(hbClass => {
+    if (!((hbClass || {}).generateTimeRef || {}).dayRef) {
```

Comment:
this is not a very clear path

sometimes its best to do it long hand  

ie 

```suggestion
    if (!hbClass.generateTimeRef || !hbClass.generateTimeRef.dayRef) {
```